services/api/src/resources/file/resolvers.ts

```diff
@@ -4,15 +4,32 @@ import { s3Client } from '../../clients/aws';
 import { query } from '../../util/db';
 import { Sql } from './sql';
 import { Sql as taskSql } from '../task/sql';
+import { s3Config } from '../../util/config';
+import { AuditLog } from '../audit/types';
+import { AuditType } from '@lagoon/commons/dist/types';
 
 // if this is google cloud storage or not
 const isGCS = process.env.S3_FILES_GCS || 'false'
 
-export const getDownloadLink: ResolverFn = async ({ s3Key }) =>
-  s3Client.getSignedUrl('getObject', {
-    Key: s3Key,
-    Expires: 300 // 5 minutes
+export const getDownloadLink: ResolverFn = async ({ s3Key }, input, { userActivityLogger }) => {
+  const auditLog: AuditLog = {
+    resource: {
+      type: AuditType.FILE,
```

Comment:
Extend this with the s3key
```suggestion
      type: AuditType.FILE,
      details: s3Key,
```

This way it will show which file was requested in the audit log. it would be nice to be able to also set the `linkedResource` to be the environment the file is attached to (same for the other two file audit logs too)
```
      "auditEvent": "api:getSignedTaskUrl",
      "emailAddress": "platformowner@example.com",
      "linkedResourceDetails": null,
      "linkedResourceId": null,
      "linkedResourceType": null,
      "resourceDetails": "tasks/130/doc-brown.jpg",
      "resourceId": null,
      "resourceType": "FILE"
```

Reply:
https://github.com/uselagoon/lagoon/issues/3933

Will deal with this in separate PR